src/string_piece.cc

```diff
@@ -0,0 +1,24 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "string_piece.h"
+
+template <>
+StringPiece::BasicStringPiece(const char* str) : str_(str), len_(strlen(str)){};
```

Comment:
there's no need for this specialization.

you can use std::char_traits<CHAR_T>::length

https://en.cppreference.com/w/cpp/string/char_traits

That does the special-casing for you.